main.go

```diff
@@ -178,6 +179,8 @@ func main() {
 		setupLog.Error(err, "problem running manager")
 		os.Exit(1)
 	}
+	// Remove all finalizers after controller is shut down
+	utils.Shutdown(mgr.GetClient())
```

Comment:
Shall we move this function before the main loop and use `defer` call instead?

Reply:
done
Adding defer before we start the manager